fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -209,17 +207,549 @@ fn expand_input_param(
+            type_field: String::from("enum unused"),
+            components: Some(vec![
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("nonexistingtype"),
+                    components: None,
+                },
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("anotherunexistingtype"),
+                    components: None,
+                },
+            ]),
+        }]);
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "Missing data: cannot parse custom type with no components"
+        )
+    }
+    #[test]
+    fn test_expand_fn_outputs_two_more_components() {
+        let result = expand_fn_outputs(&[
+            Property {
+                name: "unused".to_string(),
+                type_field: String::from("bool"),
+                components: None,
+            },
+            Property {
+                name: "unused".to_string(),
+                type_field: String::from("u64"),
+                components: None,
+            },
+            Property {
+                name: "unused".to_string(),
+                type_field: String::from("u32"),
+                components: None,
+            },
+        ]);
+        assert_eq!(result.unwrap().to_string(), "(bool , u64 , u32)");
+
+        let some_enum = Property {
+            name: "unused".to_string(),
+            type_field: String::from("enum unused"),
+            components: Some(vec![
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("nonexistingtype"),
+                    components: None,
+                },
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("anotherunexistingtype"),
+                    components: None,
+                },
+            ]),
+        };
+        let result = expand_fn_outputs(&[some_enum.clone(), some_enum]);
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "Missing data: cannot parse custom type with no components"
+        );
+
+        let some_struct = Property {
+            name: "unused".to_string(),
+            type_field: String::from("struct carmaker"),
+            components: Some(vec![
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("u64"),
+                    components: None,
+                },
+                Property {
+                    name: String::from("unused"),
+                    type_field: String::from("bool"),
+                    components: None,
+                },
+            ]),
+        };
+        let result = expand_fn_outputs(&[some_struct.clone(), some_struct]);
+        // TODO: this should panic after the function is refactored
```

Comment:
We need more context on this `TODO`. Maybe a link to the respective Github issue?

Reply:
Not even sure what I meant, removing because it will probably resurface during the refactor.